### DISCUSSION TITLE
Cannot preview markdown files in Firefox private browsing

### DISCUSSION THREAD

**Discussion by TvdW** [community/community#5278] [Codespaces]
First of all, thank you for adding VS Code directly into GitHub!

VS Code features a "Preview" button (Ctrl+Shift+V) when working with Markdown files. It works within Electron, but on the web version it gives an error.

### Steps to reproduce

1. Go to a GitHub repo (eg. https://github.com/github/feedback/)
2. Press `.`
3. Open README.md
4. Click the preview Button or press Ctrl+Shift+V

### Expected result

It should open the file in Markdown preview mode

### Actual result

    Error loading webview: Error: Could not register service workers: SecurityError: The operation is insecure..

### Environment

Firefox 93.0a1 running on Windows

**Comment by TvdW** [comment]
Further debugging shows that this is caused by Firefox's "enhanced tracking protection" mode which is enabled by default in private browsing. Turning it off allows the preview to show up.

    Failed to register/update a ServiceWorker for scope ‘https://a40d7107-e255-4d54-a7bc-d9230a1c847a.vscode-webview.net/stable/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/out/vs/workbench/contrib/webview/browser/pre/’: Storage access is restricted in this context due to user settings or private browsing mode.

It looks like Firefox disallows service workers when in private browsing mode.

**Reply by Strangehill** [reply]
this happens on Brave too, in regular browsing mode. I would guess for similar reasons.
